src/components/calculator.tsx

```diff
@@ -1,42 +1,283 @@
-import {BsFillBackspaceFill} from 'react-icons/bs'
-import {MdClear} from 'react-icons/md'
+"use client";
+
+import { BsFillBackspaceFill } from "react-icons/bs";
+import { MdClear } from "react-icons/md";
+import { TbSquareRoot2 } from "react-icons/tb";
+import { useState } from "react";
+import {
+  evaluate,
+  pi,
+  isPositive,
+  abs,
+  log,
+  factorial,
+  e,
+  sqrt,
+  sin,
+  cos,
+  tan,
+  unit,
+  ceil,
+  round,
+} from "mathjs";
+import CalcButton from "./calculatorbutton";
+
+enum CalculatorMode {
+  Standard,
+  Scientific,
+}
 
 export default function Calculator() {
+  const buttons = [7, 8, 9, "*", 4, 5, 6, "-", 1, 2, 3, "/"];
+  const [screenValue, setScreenValue] = useState("0");
+  const [calculatorMode, setCalculatorMode] = useState<CalculatorMode>(
+    CalculatorMode[0]
+  );
+  const screenClassname =
+    "h-[100px] mt-5 max-w-[300px] flex justify-end items-center pr-2 font-bold text-lg bg-white shadow-md rounded-md";
+  const screenClassnameNumIncrease =
+    "h-[100px] mt-5 max-w-[300px] flex justify-end items-center pr-2 font-bold text-sm bg-white shadow-md rounded-md";
+
+  const clearScreen = () => {
+    setScreenValue("0");
+  };
+
+  const handleBackspace = () => {
+    if (screenValue.length === 1 && screenValue !== "0") {
+      setScreenValue("0");
+    } else if (screenValue !== "0") {
+      setScreenValue((prevState) => prevState.slice(0, prevState.length - 1));
+    }
+  };
+
+  const handleButtons = (value: number | string) => {
+    if (screenValue.length !== 28) {
+      if (typeof value === "number") {
+        const numberToString = value.toString();
+
+        setScreenValue((prevState) =>
+          Number(prevState) === 0 && prevState !== "0."
+            ? numberToString
+            : prevState + numberToString
+        );
+      } else if (typeof value === "string" || value === ".") {
+        setScreenValue(screenValue + value);
+      }
+    }
+  };
+
+  const handlePi = () => {
+    if (screenValue.length !== 28) {
+      setScreenValue(pi.toString());
+    }
+  };
+
+  const handleSignInversion = () => {
+    if (screenValue.length !== 28) {
+      const valueToNumber = Number(screenValue);
+      if (isPositive(valueToNumber)) {
+        setScreenValue("-" + screenValue);
+      } else {
+        setScreenValue(abs(valueToNumber).toString());
+      }
+    }
+  };
```

Comment:
Speaking of readability, you can create a const at the top of where you defined your component 
```ts
const MAX_SCREEN_LENGTH = 28;
```
And when you're doing your evaluations it can be like
```ts
if (screenValue.length !== MAX_SCREEN_LENGTH)
```
instead of
```ts
if (screenValue.length !== 28)
```
Makes the code more readable as well.

Reply:
The reason why its in all caps is because it's a fixed value that won't ever change.
Also it should be defined outside of your component block
```ts
const MAX_SCREEN_LENGTH = 28;

export default function Calculator() {
// component code.....
}
```

Reply:
Noted. I am on it.